src/selector.rs

```diff
@@ -102,7 +102,7 @@ impl PartialEq<Ident> for SelectorSegment {
     fn eq(&self, other: &Ident) -> bool {
         match self {
             SelectorSegment::Wildcard => true,
-            SelectorSegment::Ident(ident) => ident == other,
+            SelectorSegment::Ident(ident) => other == ident,
```

Comment:
The heterogeneous PartialEq impl only exists in this direction. There isn't `String: PartialEq<Ident>`.